code/game/mecha/mecha_decal_kits.dm

```diff
@@ -0,0 +1,561 @@
+        D.decal_name = "CUSTOM: [pattern_name]"
+        D.desc = "A custom exosuit design that also overwrites the base coat of the target exosuit."
+        D.decals = list()
+        for(var/datum/mecha/mecha_decal/MD in decal_list)
+            D.decals.Add(MD.clone())
+        D.compatible_mecha = list(M.type)     // Only the exosuit this pattern was made for is compatible with a custom save.
+        D.loc = usr.loc
+        D.basecoat = current_mecha.basecoat_colour
+        D.glow = current_mecha.glow_colour
+        D.deletable = TRUE                    // Custom patterns are ALWAYS deletable.
+        to_chat(usr, "<span class='notice'>A pattern chip drops out of [src]'s input slot!</span>")
+        next_pattern_at = world.time + (30 SECONDS)
+    else
+        to_chat(usr, "<span class='warning'>The internal synthesizers haven't finished a new chip yet!</span>")
+
+// COMPATIBILITY DEFINES
+// All constructable mechs except honker and reticence.
+#define COMPATIBLE_MECHA_COMMON list(/obj/mecha/working/ripley, /obj/mecha/working/ripley/firefighter, /obj/mecha/medical/odysseus, /obj/mecha/combat/durand, /obj/mecha/combat/gygax, /obj/mecha/combat/phazon)
+// All constructable mechs
+#define COMPATIBLE_MECHA_ALL_CONSTRUCTABLE list(/obj/mecha/working/ripley, /obj/mecha/working/ripley/firefighter, /obj/mecha/medical/odysseus, /obj/mecha/combat/durand, /obj/mecha/combat/gygax, /obj/mecha/combat/phazon, /obj/mecha/combat/honker, /obj/mecha/combat/reticence)
+// All mechs (Unless you mean literally ALL mechs, don't use this)
+#define COMPATIBLE_MECHA_ALL typesof(/obj/mecha)
+// Common combat mechs
+#define COMPATIBLE_MECHA_COMBAT list(/obj/mecha/combat/durand, /obj/mecha/combat/gygax, /obj/mecha/combat/phazon)
+// Single mechs
+#define COMPATIBLE_MECHA_RIPLEY list(/obj/mecha/working/ripley, /obj/mecha/working/ripley/firefighter, /obj/mecha/working/ripley/deathripley)
+#define COMPATIBLE_MECHA_ODYSSEUS list(/obj/mecha/medical/odysseus)
+#define COMPATIBLE_MECHA_DURAND list(/obj/mecha/combat/durand)
+#define COMPATIBLE_MECHA_GYGAX list(/obj/mecha/combat/gygax)
+#define COMPATIBLE_MECHA_PHAZON list(/obj/mecha/combat/phazon)
+#define COMPATIBLE_MECHA_HONKER list(/obj/mecha/combat/honker)
+#define COMPATIBLE_MECHA_RETICENCE list(/obj/mecha/combat/reticence)
+
+////////////
+// Decals //
+////////////
+// Defines a single overlay to be applied to a mech. Icon is pulled by combining the mech's root decal string ([mech_type]-decal)
+// with the decal's decal string and a state modifier for the final form [mech_type]-decal-[decal_string] (-open, -broken).
+// 
+// A typical set of icons contains 3 states: one 4-directional one for a moving, occupied mech, one for an unoccupied open mech and 
+// one for a destroyed mech. If you do not need an icon for one or more of those states, PLEASE remember to set the corresponding 
+// has_state flag; Failing to do so will cause issues with the layering.
+//
+/datum/mecha/mecha_decal
```

Comment:
Best to split this up into another file. Rule of thumb. One file should have one object. Exceptions can be helper objects that only really exist in the context of the main object of the file.